src/rulesets/Base/mapreduce.jl

```diff
@@ -325,52 +325,37 @@ end
 ##### `cumprod`
 #####
 
-function rrule(::typeof(cumprod), x::AbstractVector{<:Real}; dims::Integer=1)
-    y = cumprod(x; dims=dims)  # does nothing unless dims == 1
-    project_x = ProjectTo(x)
-    function cumprod_pullback_1(dy_raw)
-        dy = unthunk(dy_raw)
-        dx_thunk = InplaceableThunk(
-            dx -> if dims == 1
-                ∇cumprod!(dx, x, dy, y)
-            else
-                dx .+= dy
-            end
-            ,
-            @thunk project_x(if dims == 1
-                ∇cumprod(x, dy, y)
-            else
-                dy
-            end)
-            )
-        return (NoTangent(), dx_thunk)
-    end
-    return y, cumprod_pullback_1
-end
-
-function rrule(::typeof(cumprod), x::AbstractArray{<:Real}; dims::Integer)
+function rrule(::typeof(cumprod), x::AbstractArray{<:Real}; dims::Integer=1)
```

Comment:
Is it OK to put a default value for `dims` even though there's no default value in base? https://github.com/JuliaLang/julia/blob/63e95d47e692352fac5541b6ee835b573f4c4898/base/accumulate.jl#L193

Reply:
yes, it is see:
https://github.com/JuliaDiff/ChainRules.jl/pull/784#issuecomment-1936107183